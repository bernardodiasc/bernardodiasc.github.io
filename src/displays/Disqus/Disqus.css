.Disqus {
  font-size: 1rem;
  margin: 3em 0 0;
}
