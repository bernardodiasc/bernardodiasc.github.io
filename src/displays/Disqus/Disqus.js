import React, { PureComponent } from 'react'
import { DiscussionEmbed } from 'disqus-react'
import './Disqus.css'

class Disqus extends PureComponent {
  static defaultProps = {
    url: '',
    identifier: '',
    title: '',
  }

  render() {
    const disqusShortname = 'bernardodiasc'
    const disqusConfig = {
      // url: this.props.url,
      identifier: this.props.identifier,
      title: this.props.title,
    }

    return (
      <div className="Disqus">
        <DiscussionEmbed
          shortname={disqusShortname}
          config={disqusConfig}
        />
      </div>
    )
  }
}

export default Disqus
