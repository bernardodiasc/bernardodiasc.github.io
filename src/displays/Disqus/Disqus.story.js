import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import Disqus from './Disqus'
import DisqusReadme from './README.md'
storiesOf('Displays/Disqus', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{DisqusReadme}</Docs>)
  .add('Example', () => <Disqus />)
  .add('Minimal render', () => <Disqus />)
