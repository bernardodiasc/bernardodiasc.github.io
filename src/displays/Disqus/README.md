# Disqus

## How to use

```
import Disqus from 'displays/Disqus'
```

```
<Disqus />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
