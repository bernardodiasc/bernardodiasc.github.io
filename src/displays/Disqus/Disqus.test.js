import React from 'react'
import { shallow } from 'enzyme'

import Disqus from './Disqus.js'

it('renders without crashing', () => {
  const component = shallow(<Disqus />)
  expect(component).toMatchSnapshot()
})
