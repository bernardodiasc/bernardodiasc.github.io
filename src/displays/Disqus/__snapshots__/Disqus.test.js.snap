// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Disqus"
>
  <DiscussionEmbed
    config={
      Object {
        "identifier": "",
        "title": "",
      }
    }
    shortname="bernardodiasc"
  />
</div>
`;
